### PR TITLE
Selenium Task 40

### PR DESCRIPTION
**Pull request by alexcrtCS**
- Solved all exercises from list for Task 40
- Each solution in separate class

### CODE REVIEW

File: src/test/java/ParameterizedLoginTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {USERNAME, USERNAME2})
+    void loginWithAccountsTest(String userId) throws InterruptedException {
+        driver.get(WEBPAGE);
+        driver.findElement(MAIL_LOGIN).click();
+        driver.findElement(USER_FIELD).sendKeys(userId);
+        driver.findElement(SIGN_IN_BTN).click();
+        driver.findElement(PASS_FIELD).sendKeys(PASSWORD);
```
Comment (irynaalex) [inline src/test/java/ParameterizedLoginTest.java:44]: In the task we have: "with at least 2 different credentials", so it should be pair  (userName, password)

File: src/test/java/ParameterizedLoginTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        // Must pause before submitting details to prevent detection of Selenium usage
+        Thread.sleep(6000); // at least 5s are required to pass, else Selenium is flagged as suspicious activity
```
Comment (irynaalex) [inline src/test/java/ParameterizedLoginTest.java:46]: What type of waiter is it (add your answer as comment near sleep)?

File: src/test/java/EmployeeTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import org.openqa.selenium.support.ui.Select;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EmployeeTest {
+    private final String WEBPAGE = "https://demo.seleniumeasy.com/table-sort-search-demo.html";
+    private final By ENTRY_SELECT = By.name("example_length");
+    private final By NEXT_BTN = By.id("example_next");
+    private final By TABLE_ROWS = By.xpath("//tr");
```
Comment (LikBez123) [inline src/test/java/EmployeeTest.java:21]: your locator also captures the header of the table, which is not needed

File: src/test/java/EmployeeTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    String position = row.findElement(POSITION_CELLS).getText();
+                    String office = row.findElement(OFFICE_CELLS).getText();
+                    // Will take all relevant details and create new employee then add it to our list
+                    Employee employee = new Employee(name, position, office);
+                    employeeList.add(employee);
+                }
+            });
+            // Clicking next to get the following 10 rows of our table
+            driver.findElement(NEXT_BTN).click();
+        } while (!driver.findElement(NEXT_BTN).getAttribute("class").endsWith("disabled"));
```
Comment (LikBez123) [inline src/test/java/EmployeeTest.java:69]: the approach is correct, but you will not collect data for the last pagination result page. By clicking the last time, you will immediately get button disabled without collecting data from the last page. Try this:
```
 boolean enabled = true;
        while (enabled) {
            WebElement nextBtn = driver.findElement(NEXT_BTN);
            List<WebElement> rowList = driver.findElements(TABLE_ROWS);
            for (WebElement row : rowList) {
                int actualAge = Integer.parseInt(row.findElement(AGE_CELLS).getText());
                int actualSalary = Integer.parseInt(row.findElement(SALARY_CELLS).getAttribute("data-order"));
                if (actualAge > minAge && actualSalary <= maxSalary) {
                    String name = row.findElement(NAME_CELLS).getText();
                    String position = row.findElement(POSITION_CELLS).getText();
                    String office = row.findElement(OFFICE_CELLS).getText();
                    Employee employee = new Employee(name, position, office);
                    employeeList.add(employee);
                }
            }
            if (!nextBtn.getAttribute("class").contains("disabled")) {
                nextBtn.click();
            } else {
                enabled = false;
            }
        }
```

File: src/test/java/UserTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    @BeforeEach
+    void initChromeDriver() {
+        driver = new ChromeDriver();
+        driver.get(WEBPAGE);
+        driver.manage().window().maximize();
+    }
+
+    // Simple test for checking that user image is displayed
+    @Test
+    void UserImageTest() {
```
Comment (LikBez123) [inline src/test/java/UserTest.java:39]: the name of the methods begins with lowercase. Check all the tests

**Review by LikBez123 — CHANGES_REQUESTED** [review]

**Review by LikBez123 — APPROVED** [review]

**Pull request merged**